src/codeview.js

```diff
@@ -224,7 +224,7 @@ var Codeview = GObject.registerClass({
                     this._buffer.text = value;
             });
             this._cached_ast = null;
-            this.setCompileResults([]);
+            this.setCompileResults([], false);
```

Comment:
I think the assumption under which I originally wrote this line (that programmatically-set text will always have blank compile results) no longer holds. I think it would be good to instead call `this.compile()` here (and that means you cuold remove `this._compile()` from the constructor of `UserFunction`), but that would require adding the boolean flag as a parameter to the `should-compile` signal as suggested above.